lib/mail/attachments_list.rb

```diff
@@ -71,6 +72,7 @@ def []=(name, value)
 
       if hash[:body].respond_to? :force_encoding and hash[:body].respond_to? :valid_encoding?
         if not hash[:body].valid_encoding? and default_values[:content_transfer_encoding].downcase == "binary"
+          hash[:body] = hash[:body].dup if hash[:body].frozen?
           hash[:body].force_encoding("BINARY")
```

Comment:
This is in the gems code as opposed to test, and I figured the body could be large -- for performance and memory reasons I didn't want to dup it unless necessary